### PR TITLE
Bug: PouchDB Strangeness

### PR DESCRIPTION
**Pull request by swashcap**
- **Problem:**
  1. PouchDB Server’s teardown was convoluted
  2. `DBListener#destroy` is actually _async_ (due to Pouch internals), but we were treating it as _sync_
- **Solution:**
  1. Bump pouch and drop unneeded Bluebird delay
  2. Return a promise on `DBListener#destroy`. Listen for the `complete` event on the PouchDB `Changes` event emitter and resolve when fired.
- **Testing:** Run tests in affected code (_coinstac-common_ and _coinstac-client-core_)

### CODE REVIEW

File: packages/coinstac-common/src/models/db-listener.js [hunk 1]
```diff
@@ -42,8 +42,11 @@ class DBListener extends EventEmitter {
    */
   destroy() {
     this.removeAllListeners();
-    this.changes.removeAllListeners();
-    this.changes.cancel();
+
+    return new Promise(resolve => {
+      this.changes.on('complete', resolve);
```
Comment (swashcap) [inline packages/coinstac-common/src/models/db-listener.js:47]: This is _totally_ undocumented. PouchDB’s `Cancel` emitter fires a `cancel` event. Something _somewhere_ catches this and emits a `complete` event when all requests have finished.

File: packages/coinstac-common/test/models/pipeline/runner/pool/.test-pool-utils.js [hunk 2]
```diff
@@ -249,14 +249,10 @@ module.exports = {
     },
 
     teardown: function() {
-      // https://github.com/hoodiehq/spawn-pouchdb-server/issues/41
```
Comment (swashcap) [inline packages/coinstac-common/test/models/pipeline/runner/pool/.test-pool-utils.js:252]: Props to @cdaringe on this one.

File: packages/coinstac-server-core/package.json [hunk 3]
```diff
@@ -45,7 +45,6 @@
     "jsdoc": "^3.4.0",
     "minami": "^1.1.1",
     "nyc": "^6.4.4",
-    "pouchy": "^9.0.2",
```
Comment (swashcap) [inline packages/coinstac-server-core/package.json:48]: This is never used by _coinstac-server-core_ as we’re relying on _coinstac-common_ for all DB goodness.

File: packages/coinstac-common/src/models/db-listener.js (hunk 1 above)
Comment (cdaringe) [inline packages/coinstac-common/src/models/db-listener.js:47]: Added a doc PR to pouchdb

File: packages/coinstac-common/test/models/pipeline/runner/pool/.test-pool-utils.js (hunk 2 above)
Comment (cdaringe) [inline packages/coinstac-common/test/models/pipeline/runner/pool/.test-pool-utils.js:252]: Kinda :$. More daleharvey :). I guess I asked loudly enough!

**Pull request merged**